### DISCUSSION TITLE
Is it possible to use XState simply as a UI access layer?

### DISCUSSION THREAD

**Discussion by toddtarsi** [statelyai/xstate#5065] [Q&A]
Basically, the visualizer looks great, and we want to have a capability for us and our clients to drag and drop some state stuff and view our subset of workflow steps, but for us here, that's really just a series of PG tables on our backend. Is it possible / reasonable to instrument xstate as a client facing tool, or is it kinda inseparable from stately on the backend?

**Comment by davidkpiano** [comment]
Right now, it's pretty tied to Stately on the backend. You will soon be able to self-host Stately Studio though, but if you just need the visual layer, I can consider making just that available for self-hosting as well.

**Reply by toddtarsi** [reply]
No pressure, as I know its hard as heck to do Github in evenings. I've honestly been on an OSS hiatus for a couple months.

But I definitely would love to use your amazing visualizer / editor as a design tool. It looks so darn good! When you announce self hosting, can you ping me?

**Reply by toddtarsi** [reply]
Actually, I'll star the repo. I think that will subscribe me to it somehow 😆 Thanks again!
